src/eval_iter_loop.ipynb

```diff
@@ -0,0 +1,323 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# imports\n",
+    "from fondant.pipeline import ComponentOp, Pipeline\n",
+    "from pathlib import Path\n",
+    "\n",
+    "from fondant.pipeline.runner import DockerRunner\n",
+    "import pipeline_index, pipeline_eval"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Single run evaluation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# get Host IP address\n",
+    "import socket\n",
+    "\n",
+    "def get_host_ip():\n",
+    "    try:\n",
+    "        # Create a socket object and connect to an external server (e.g., Google's public DNS server)\n",
+    "        # This step is done to get the local machine's IP address\n",
+    "        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)\n",
+    "        s.connect((\"8.8.8.8\", 80))\n",
+    "        host_ip = s.getsockname()[0]\n",
+    "    except Exception as e:\n",
+    "        print(f\"Error while retrieving host IP address: {e}\")\n",
+    "        host_ip = None\n",
+    "    finally:\n",
+    "        s.close()\n",
+    "\n",
+    "    return host_ip\n",
+    "\n",
+    "# Example usage\n",
+    "host_ip = get_host_ip()\n",
+    "print(f\"Host IP address: {host_ip}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fixed_args = {\n",
+    "    \"pipeline_dir\":\"./data-dir\",\n",
+    "    \"embed_model_provider\":\"huggingface\",\n",
+    "    \"embed_model\":\"all-MiniLM-L6-v2\",\n",
+    "    \"weaviate_url\":f\"http://{host_ip}:8080\", # IP address \n",
+    "    \"weaviate_class_name\":\"Pipeline1\",\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# set-up the indexing pipeline\n",
+    "indexing_args = {\n",
+    "    \"hf_dataset_name\":\"wikitext@~parquet\",\n",
+    "    \"data_column_name\":\"text\",\n",
+    "    \"n_rows_to_load\":1000,\n",
+    "    \"chunk_size\":512,\n",
+    "    \"chunk_overlap\":32\n",
+    "}\n",
+    "\n",
+    "indexing_pipeline = pipeline_index.create_pipeline(**fixed_args, **indexing_args)\n",
+    "\n",
+    "# indexing_pipeline = pipeline_index.create_pipeline(\n",
+    "#     pipeline_dir=\"./data-dir\",\n",
+    "#     embed_model_provider=\"huggingface\",\n",
+    "#     embed_model=\"all-MiniLM-L6-v2\",\n",
+    "#     weaviate_url=f\"{host_ip}:8080\", # IP address \n",
+    "#     weaviate_class_name=\"Pipeline_1\",\n",
+    "#     hf_dataset_name=\"wikitext@~parquet\",\n",
+    "#     data_column_name=\"text\",\n",
+    "#     n_rows_to_load=1000,\n",
+    "#     chunk_size=512,\n",
+    "#     chunk_overlap=32\n",
+    "# )"
+   ]
```

Comment:
@RobbeSneyders would you prefer to create the pipeline using kwargs as it is, or do it as it is commented? I find the kwargs cleaner but the second option may be better if the user wants to modify the pipeline. What do you think?